src/shared/sol-util.c

```diff
@@ -905,3 +905,79 @@ sol_util_uuid_str_valid(const char *str)
 
     return true;
 }
+
+SOL_API int
+sol_util_escape_quotes(const struct sol_str_slice slice,
+    struct sol_buffer *buf)
+{
+    bool is_escaped = false;
+    size_t i, last_append;
+    int r;
+
+    SOL_NULL_CHECK(buf, -EINVAL);
+
+    sol_buffer_init(buf);
+
+    if (!slice.len)
+        return 0;
+
+    last_append = 0;
+
+    for (i = 0; i < slice.len; i++) {
+        if (!is_escaped && (slice.data[i] == '"' || slice.data[i] == '\'')) {
```

Comment:
@barbieri 
In this case x'y', I prefer to return xy, instead of just returning x or an error.
Is that ok for you?